components/function-controller/internal/controllers/serverless/system_state.go

```diff
@@ -136,13 +136,25 @@ var (
 	optionalTrue = true
 )
 
+func (s *systemState) getResourceRequirements() corev1.ResourceRequirements {
```

Comment:
I would remove this funciton from `systemState` to not transform `systemState` to `god` object.
You can pass function as argument.
I would name it `getBuildResourceRequirements`.

Reply:
Done